spanner/grpc_client.go

```diff
@@ -65,10 +70,43 @@ type spannerClient interface {
 // grpcSpannerClient is the gRPC API implementation of the transport-agnostic
 // spannerClient interface.
 type grpcSpannerClient struct {
+	id                   uint64
 	raw                  *vkit.Client
 	metricsTracerFactory *builtinMetricsTracerFactory
+
+	// These fields are used to uniquely track x-goog-spanner-request-id
+	// grpc.ClientConn is presumed to be the channel, hence channelID
```

Comment:
The `raw *vkit.Client` is the channel, so in that sense, this could be said to be redundant. But that property does not have a simple number or other simple string representation, which means that it is probably better/easier to just use the channel pool index that was used to fetch the channel as the channel ID here. Which again means that this property is not redundant and should be assigned a value.

Reply:
Yeah I revised the design entirely and with a much better and simpler functionality whereby each channelID is retrieved from the pool, each clientID is retrieved from the globally incremented value so this should no longer be a problem.